server/app/models/Application.java

```diff
@@ -52,6 +53,17 @@ public class Application extends BaseModel {
   private String latestStatus;
   private boolean isAdmin;
 
+  private static ImmutableMap<LifecycleStage, String> submissionStatus =
+      ImmutableMap.of(
+          LifecycleStage.DRAFT,
+          "NOT_SUBMITTED",
+          LifecycleStage.ACTIVE,
+          "CURRENT",
+          LifecycleStage.OBSOLETE,
+          "OBSOLETE",
+          LifecycleStage.DELETED,
+          "DELETED");
+
```

Comment:
This should get encapsulated into it's own method on the `LifecycleStage` enum. Like a `getDescription` method or something of that ilk. 

```
LifecycleStage stage  = LifecycleStage.DRAFT;
String description = stage.getDescription();
```

Reply:
You know, I had it this way before and then moved it to Application, because I thought the submission status seemed related to the application. But looking at your suggestion now, I agree. Done!